### ISSUE TITLE
Selenium::WebDriver::Error::SessionNotCreatedError

### ISSUE THREAD

**Issue by shamas-** [heroku/heroku-buildpack-chrome-for-testing#11]
I figured this might be a drop-in replacement for "heroku/google-chrome" and "heroku/chromedriver" as "heroku-community/chrome-for-testing." Is it supposed to be? I tried and I'm getting errors:

```
Selenium::WebDriver::Error::SessionNotCreatedError: session not created: Chrome failed to start: exited normally.
  (session not created: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /app/.chrome-for-testing/chrome-linux64/chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
```

Help?

**Comment by mars** [comment]
Have you attempted any diagnosis?

I just searched the internet for the error, and found a couple StackOverflow posts ([1](https://stackoverflow.com/questions/56637973/how-to-fix-selenium-devtoolsactiveport-file-doesnt-exist-exception-in-python), [2](https://stackoverflow.com/questions/50642308/webdriverexception-unknown-error-devtoolsactiveport-file-doesnt-exist-while-t)) with `chrome` flag solutions.

I see that the `google-chrome` buildpack does [fill in some default flags](https://github.com/heroku/heroku-buildpack-google-chrome/blob/master/bin/compile#L182):

```
--headless --no-sandbox --disable-gpu --remote-debugging-port=9222
```

During our months of testing with this new buildpack, we didn't experience this issue, so the `chrome` command provided by this buildpack currently does not wrap the executable with these defaults. I was concerned that the evolution of Chrome, such as [the `--headless=new` variation](https://developer.chrome.com/docs/chromium/new-headless) would potentially conflict with those default assumptions.

**Comment by mars** [comment]
What language/framework/tool are you using with Chrome for Testing?

**Comment by alvesian** [comment]
Just for information. I was with this problem using python. But my options.add_argument was ('--headless=new'). When I removed and added only @mars flags, worked like a charm! Thanks

**Comment by shamas-** [comment]
> Have you attempted any diagnosis?

I have. I didn't want to bias your response in case I was missing something easy so I didn't mention anything up front. 🙂

> What language/framework/tool are you using with Chrome for Testing?

Ruby, Ruby on Rails, system tests with a usual Capybara & Selenium combo.

I have since been able to get tests to fire up with this new buildpack. I added all of the flags you mentioned and things worked. From there I whittled it down to these flags being the necessary ones:

- `headless` (`--headless=new` doesn't consistently work for me)
- `no-sandbox`

With the change to this buildpack I have been able to remove these (for now):
-  `disable-dev-shm-usage`
- assigning Chrome's binary path via [the previous package's `GOOGLE_CHROME_SHIM`](https://github.com/ashwinstr/heroku-buildpack-google-chrome?tab=readme-ov-file#selenium)

I'm not sure if you're into elaborating on migration instructions for this, but it could be nice to save future people some hours.
